src/client/app/actions/snippet.ts

```diff
@@ -54,7 +54,7 @@ export class ViewAction implements Action {
 export class ImportAction implements Action {
     readonly type = SnippetActionTypes.IMPORT;
 
-    constructor(public mode: string, public payload?: string, public isViewMode?: boolean) { }
+    constructor(public mode: string, public payload: { data: string, isViewMode: boolean }) { }
```

Comment:
I think the point of having the single payload is that "mode" should be part of it too.  I.e., ImportAction, just like any other action, should have only a single argument in the constructor, `payload`, which should contain all the info it ever needs.